lib/autoPagination.js

```diff
@@ -21,6 +22,10 @@ function makeAutoPaginationMethods(self, requestArgs, spec, firstPagePromise) {
       );
     }
 
+    if (i === 0 && reverseIteration) {
+      listResult.data.reverse();
```

Comment:
We should avoid mutating the list. Can use "reverseIteration" to index from the end, instead? The mutation can leak in subtle ways, e.g.:
```javacript
      const req = stripe.customers.list({
        limit: 2,
        email,
        ending_before: midPointCustomerId,
      });
      const firstTwoCusts = await req;
      console.log(firstTwoCusts.data.map((x) => x.id)); // cus_3, cus_4
      const moreCusts = await req.autoPagingToArray({limit: 4});
      console.log(moreCusts.map((x) => x.id)); // cus_4, cus_3, cus_2, cus_1
      console.log(firstTwoCusts.data.map((x) => x.id)); // cus_4, cus_3
```

Reply:
@richardm-stripe thanks for the review, and much thanks for pointing this out (it didn't occur to me)

i'll see to implement this - thinking of closure around the index to save us the conditional-branching on the `reverseIteration` on the each iteration

Reply:
made a simple numeric index manipulation, as suggested (91c78e9); adding another array iterator closure feels redundant and awkwardly similar to existing `iterate` function.